framework/src/messaging/messaging_recv.c

```diff
@@ -0,0 +1,384 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <debug.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <mqueue.h>
+#include <stdlib.h>
+#include <signal.h>
+#include <string.h>
+#include <unistd.h>
+#include <queue.h>
+#include <sys/types.h>
+#include <messaging/messaging.h>
+#include "messaging_internal.h"
+
+#define MAX_PORT_NAME_SIZE 64
+sq_queue_t g_port_info_list;
+struct msg_port_info_s {
+	struct msg_port_info_s *flink;
+	char name[MAX_PORT_NAME_SIZE];
+	mqd_t mqdes;
+};
+typedef struct msg_port_info_s msg_port_info_t;
+
+/****************************************************************************
+ * private functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name : messaging_set_notification
+ * 
+ * Description:
+ *  This function calls mq_notify to set message queue notification.
+ ****************************************************************************/
+static int messaging_set_notification(messaging_recv_info_t *data)
+{
+	int ret;
+	struct sigevent notification;
+
+	notification.sigev_notify = SIGEV_SIGNAL;
+	notification.sigev_signo = SIGMSG_UNICAST;
+	notification.sigev_value.sival_ptr = data;
+
+	ret = mq_notify(data->mqdes, &notification);
+	if (ret < 0) {
+		msgdbg("message recv fail : set notification fail. %d\n", errno);
+		MSG_FREE(data);
+		return ERROR;
+	}
+	return OK;
+}
+
+/****************************************************************************
+ * Name : messaging_nonblock_callback
+ * 
+ * Description:
+ *  This function is wrapper callback function for waiting until receiving the message.
+ ****************************************************************************/
+static void messaging_nonblock_callback(int signo, siginfo_t *data)
```

Comment:
nonblock_callback what? run? register? Please add a verb to say what this function is exactly.